content/en/llm_observability/setup/sdk/nodejs.md

```diff
@@ -0,0 +1,713 @@
+
+`mlApp`
+: optional - _string_
+<br />The name of your LLM application, service, or project, under which all traces and spans are grouped. This helps distinguish between different applications or experiments. See [Application naming guidelines](#application-naming-guidelines) for allowed characters and other constraints. To override this value for a given trace, see [Tracing multiple applications](#tracing-multiple-applications). If not provided, this defaults to the value of `DD_LLMOBS_ML_APP`.
+
+`agentlessEnabled`
+: optional - _boolean_ - **default**: `false`
+<br />Only required if you are not using the Datadog Agent, in which case this should be set to `true`. This configures the `ddtrace` library to not send any data that requires the Datadog Agent. If not provided, this defaults to the value of `DD_LLMOBS_AGENTLESS_ENABLED`.
+
+`apiKey`
+: optional - _string_ 
+<br />Your Datadog API key. If not provided, this defaults to the value of `DD_API_KEY`.
+
+These options can be set on the general tracer configuration:
+
+`site`
+: optional - _string_ 
+<br />The Datadog site to submit your LLM data. Your site is {{< region-param key="dd_site" code="true" >}}. If not provided, this defaults to the value of `DD_SITE`.
+
+`env`
+: optional - _string_
+<br />The name of your application's environment (examples: `prod`, `pre-prod`, `staging`). If not provided, this defaults to the value of `DD_ENV`.
+
+`service`
+: optional - _string_
+<br />The name of the service used for your application. If not provided, this defaults to the value of `DD_SERVICE`.
+
+### AWS Lambda setup
+
+Use the `llmobs.flush()` function to flush all remaining spans from the tracer to LLM Observability at the end of the Lambda function.
+
+#### Application naming guidelines
+
+Your application name (the value of `DD_LLMOBS_ML_APP`) must be a lowercase Unicode string. It may contain the characters listed below:
+
+- Alphanumerics
+- Underscores
+- Minuses
+- Colons
+- Periods
+- Slashes
+
+The name can be up to 193 characters long and may not contain contiguous or trailing underscores.
+
+## Tracing spans
+
+To trace a span, use `llmobs.wrap(spanKind, options, function)` as a function wrapper for the function you'd like to trace. For a list of available span kinds, see the [Span Kinds documentation][8]. For more granular tracing of operations within functions, see [Tracing spans using inline methods](#tracing-spans-using-inline-methods).
+
+### Automatic argument, output, and function name Capturing
+
+`llmobs.wrap` (along with [`llmobs.decorate`](#function-decorators-in-typescript)) will try to automatically capure inputs, outputs, and the name of the function being traced. If you need to manually annotate a span, see [Annotating a span](#annotating-a-span). Inputs and outputs you annotate will override the automatic capturing. Additionally, to override the function name, pass the `name` property on the options object to the `llmobs.wrap` function:
```

Comment:
```suggestion
`llmobs.wrap` (along with [`llmobs.decorate`](#function-decorators-in-typescript) for typescript) will try to automatically capure inputs, outputs, and the name of the function being traced. If you need to manually annotate a span, see [Annotating a span](#annotating-a-span). Inputs and outputs you annotate will override the automatic capturing. Additionally, to override the function name, pass the `name` property on the options object to the `llmobs.wrap` function:
```
Should mention typescript here, otherwise people might get a bit confused by the second function name dropping